spec/models/upload_spec.rb

```diff
@@ -47,6 +47,156 @@
     end
   end
 
+  describe "video conversion" do
+    let(:user) { Fabricate(:user) }
+
+    before do
+      # Add mp4 to authorized extensions for video uploads
+      extensions = SiteSetting.authorized_extensions.split("|")
+      SiteSetting.authorized_extensions = (extensions | ["mp4"]).join("|")
+
+      SiteSetting.video_conversion_service = "aws_mediaconvert"
+      SiteSetting.mediaconvert_role_arn = "arn:aws:iam::123456789012:role/MediaConvertRole"
+      SiteSetting.enable_s3_uploads = true
+      SiteSetting.s3_use_iam_profile = true
+      SiteSetting.video_conversion_enabled = true
+      allow(Jobs).to receive(:enqueue)
+    end
+
+    context "when video conversion is enabled" do
+      it "enqueues a convert_video job for supported video files on create" do
+        allow(FileHelper).to receive(:is_supported_video?).with("small.mp4").and_return(true)
+
+        upload = Fabricate(:upload, original_filename: "small.mp4", extension: "mp4", user: user)
+
+        expect(Jobs).to have_received(:enqueue).with(:convert_video, upload_id: upload.id)
```

Comment:
We have helper methods in [SidekiqHelpers](https://github.com/discourse/discourse/blob/a3018bafa9814ec4e247ab0c0e8b361b3bed745c/spec/support/sidekiq_helpers.rb) that helps you to check if the right job has been enqueued.